tests/std/tests/GH_004108_some_ranges_algos_construct_wrong_type/test.compile.pass.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <ranges>
+
+using namespace std;
+
+namespace gh4108 {
+    struct RRef {
+        RRef(RRef&&) = delete;
+    };
+
+    struct Val {
+        Val(const RRef&);
+        auto operator<=>(const Val&) const = default;
+    };
+
+    struct I {
+        using value_type      = Val;
+        using difference_type = ptrdiff_t;
+        Val& operator*() const;
+        I& operator++();
+        I operator++(int);
+        I& operator--();
+        I operator--(int);
+        bool operator==(const I&) const;
+        friend RRef&& iter_move(const I&);
+    };
+
+    // GH-4108 <algorithm>: ranges::inplace_merge accidentally constructed remove_cvref_t<iter_rvalue_reference_t <I>>

```

Comment:
This faithfully replicated a weird space in the issue's title, `iter_rvalue_reference_t <I>`. I've fixed the issue and will fix the comment.